curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a23c1d505bfa13747c8a9b.md

```diff
@@ -34,7 +34,7 @@ Your first nested `span` element should have the `id` of `xpText`.
 const stat = document.querySelectorAll('.stat')[0];
 const strong = stat?.querySelector('strong');
 const span = strong?.querySelector('span');
-assert.equal(span?.id, 'xpText');
+assert.match(span?.id, /xpText/);
```

Comment:
I don't think we want `.match` on the ID steps, because the ID needs to be exact.

Reply:
you are right but one thing I am also seeing is If we use` assert.match` on `innerText` like this` assert.match(stat.innerText, /XP: 0/);` it will also allow user to pass test case with any string which contains the matching string.
If that is fine, I mean I f we can allow user to have any `innerText` unless it contains matching string to pass test cases, please let me know I will make that changes.
![ss2](https://github.com/freeCodeCamp/freeCodeCamp/assets/32094231/ed399cec-395f-4b91-9b7d-c89159a3d57b)

Reply:
We could assert that it matches the full string, and prefix the regex with optional space classes.

Reply:
Sure I will do that
Thanks